### PR TITLE
Issue 543 - Catalog endpoint does not return usage info

### PR DESCRIPTION
**Pull request by javier-gomez-hs**
- Added Usage Info to Phase in Catalog
- Added Integration Test to verify the behavior
- Updated OSS-Parent version to match KB Client changes

### CODE REVIEW

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/json/CatalogJson.java [hunk 1]
```diff
@@ -375,22 +426,204 @@ public int hashCode() {
         }
     }
 
+    public static class TieredBlockJson {
```
Comment (pierre) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/json/CatalogJson.java]: Can you try to run the [Ruby code generator](https://github.com/killbill/killbill-java-parser) (see the [example](https://github.com/killbill/killbill-client-ruby/blob/master/gen_config/model.conf))?

This generator takes care of sync'ing the JAX-RS models to their Ruby equivalent (https://github.com/killbill/killbill-client-ruby/tree/master/lib/killbill_client/models/gen). We've had sometimes issues in the past where inner classes would confuse the tool. No need to check-in anything / creating a new PR, let's just make sure the Ruby classes are properly generated.

Thanks!

Comment (javier-gomez-hs) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/json/CatalogJson.java]: @pierre 
I've cloned the **killbill-java-parser** project and tried with this command:
```
java -cp target/killbill-java-parser-0.0.1-SNAPSHOT.jar com.ning.killbill.KillbillParser -i file:///$KILLBILL_REPO/jaxrs/src/main/java/ -o /tmp/output -t RUBY_CLIENT_API -p org.killbill.billing.jaxrs.json
```

Replacing the $KILLBILL_REPO variable with my location. But I got this error:
```
Error: Could not find or load main class com.ning.killbill.KillbillParser
```

I've tried to compile the project using `mvn clean install` but there is an error:
```
[ERROR] Failed to execute goal org.antlr:antlr4-maven-plugin:4.0:antlr4 (default) on project killbill-java-parser: Execution default of goal org.antlr:antlr4-maven-plugin:4.0:antlr4 failed: Index: 0, Size: 0 -> [Help 1]
```

Do I have to clone / import anything else?

Thanks!!

Comment (pierre) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/json/CatalogJson.java]: Are you on JDK8? Could you switch to JDK7 and retry to build it?

See https://github.com/antlr/antlr4/issues/494.

Comment (javier-gomez-hs) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/json/CatalogJson.java]: @pierre 
Using JDK7 I was able to generate the files!
Do I have to do anything else? Push any change?
I've checked that the new JAX-RS models are now generated in Ruby.

Thanks!

Comment (sbrossie) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/json/CatalogJson.java]: @javier-gomez-hs I think this is good enough.

**Comment by sbrossie** [comment]
👍 Looks goos to me. @pierre Are you ok, with a merge?

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/json/CatalogJson.java [hunk 2]
```diff
@@ -375,22 +439,349 @@ public int hashCode() {
...
+        private final String unit;
+        private final String size;
+        private final String max;
+        private final List<PriceJson> prices;
+
+        @JsonCreator
+        public TieredBlockJson(@JsonProperty("billingPeriod") final String unit,
```
Comment (pierre) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/json/CatalogJson.java]: ⚠️ Typo: `unit`

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/json/CatalogJson.java [hunk 3]
```diff
@@ -375,22 +439,349 @@ public int hashCode() {
...
+        }
+    }
+
+    public static class LimitJson {
+        private final UnitJson unit;
+        private final String max;
+        private final String min;
+
+        @JsonCreator
+        public LimitJson(@JsonProperty("unit") final UnitJson unit,
```
Comment (pierre) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/json/CatalogJson.java]: Do we really need `UnitJson`? Above, in `TieredBlockJson`, it's just a String.

**Comment by pierre** [comment]
Just added a couple of remarks.

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/json/CatalogJson.java (hunk 3 above)
Comment (javier-gomez-hs) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/json/CatalogJson.java]: @pierre 
We probably don't need the UnitJson object!
I've removed it.

Thanks!

**Comment by javier-gomez-hs** [comment]
@pierre @sbrossie 
I've added the suggested changes.

Thanks!!

**Pull request merged**